crates/subspace-farmer/src/dsn/tests.rs

```diff
@@ -122,3 +136,207 @@ async fn no_sync_test() {
+            Plot::open_or_create(base_path, public_key, max_piece_count)
+        };
+
+        let mut syncer_multi_farming = MultiFarming::new(
+            MultiFarmingOptions {
+                base_directory: syncer_base_directory.as_ref().to_owned(),
+                archiving_client: syncer_client.clone(),
+                farming_client: syncer_client.clone(),
+                object_mappings: object_mappings.clone(),
+                reward_address: subspace_core_primitives::PublicKey::default(),
+                bootstrap_nodes: vec![seeder_multiaddr.clone()],
+                listen_on: vec!["/ip4/127.0.0.1/tcp/0".parse().unwrap()],
+                dsn_sync: false,
+            },
+            syncer_max_plot_size * PIECE_SIZE as u64,
+            syncer_max_plot_size,
+            plot_factory,
+            false,
+        )
+        .await
+        .unwrap();
+
+        let node_runners = futures::future::join_all(
+            std::mem::take(&mut syncer_multi_farming.networking_node_runners)
+                .into_iter()
+                .map(NodeRunner::run),
+        );
+        tokio::spawn(async move {
+            node_runners.await;
+        });
+
+        let range_size = PieceIndexHashNumber::MAX / seeder_max_plot_size * request_pieces_size;
+        let mut futures = syncer_multi_farming
+            .single_plot_farms
+            .iter()
+            .map(|farming| {
+                let plot = farming.plot.clone();
+                farming
+                    .dsn_sync(syncer_max_plot_size, seeder_max_plot_size, range_size)
+                    .map(move |result| {
+                        result.unwrap();
+                        plot
+                    })
+            })
+            .collect::<FuturesUnordered<_>>();
+
+        while let Some(plot) = futures.next().await {
+            let sync_sector_size =
+                PieceIndexHashNumber::MAX / seeder_max_plot_size * syncer_max_plot_size;
+            let expected_start =
+                U256::from_big_endian(&plot.public_key()).wrapping_sub(&(sync_sector_size / 2));
+            let expected_end =
+                U256::from_big_endian(&plot.public_key()).wrapping_add(&(sync_sector_size / 2));
+            let public_key = U256::from_big_endian(&plot.public_key());
+            let Range { start, end } = plot.get_piece_range().unwrap();
+
+            let shift_to_middle =
+                |n: U256, pub_key| n.wrapping_sub(pub_key).wrapping_add(&U256::MIDDLE);
+
+            let expected_start = shift_to_middle(expected_start, &public_key);
+            let expected_end = shift_to_middle(expected_end, &public_key);
+            let start = shift_to_middle(U256::from_big_endian(&start.0), &public_key);
+            let end = shift_to_middle(U256::from_big_endian(&end.0), &public_key);
+
+            assert!(expected_start <= start && start <= expected_end);
+            assert!(expected_start <= end && end <= expected_end);
```

Comment:
This to looks like a very implementation-driven test that doesn't even check what was synced, just the range.

It might be helpful, but I had different expectations. @shamil-gadelshin WDYT about the check here?

Reply:
This is likely helpful but I would prefer having more strict checks of the synchronized content. There is a way to calculate an expected sync result for the random 'syncer' public key because we control the pieces production in this mock.